### PR TITLE
Projectors: add the ability for multiple projectors (up to 4) to proj…

### PR DESCRIPTION
**Pull request by gwaldron**
…ect onto node

Run example_projectors and go to '2' to test.

### CODE REVIEW

File: SDK/simVis/Projector.cpp [hunk 1]
```diff
@@ -132,30 +132,143 @@ static const float DEFAULT_ALPHA_VALUE = 0.1f;
...
-      cv->pushStateSet(ss.get());
-      traverse(node, nv);
-      cv->popStateSet();
+      ProjectorObservers keep;
+      for (auto& projector : projectors_)
+      {
+        if (projector.valid())
+          keep.push_back(projector);
+      }
+      projectors_.swap(std::move(keep));
```
Comment (emminizer) [inline SDK/simVis/Projector.cpp:193]: I am getting a compile error with this line on MSVC 2019. I replaced locally with `projectors_ = keep;`

**Comment by gwaldron** [comment]
Can you try:
`projectors_.swap(keep);`

The move semantics should be built into the swap function anyway; it will be more efficient than an assignment.

**Comment by gwaldron** [comment]
I am not seeing those warnings. Are you on an NVIDIA card?
I did have to update ArrayUniform to support vec4's, in f9c0f26 .. did you get that update?

**Comment by emminizer** [comment]
Yes I am on an NVIDIA card. I am on current master and do have f9c0f26.

`ArrayUniform::ensureCapacity()` has no ability to handle vec4, so it retrieves and sets as a float.

**Comment by gwaldron** [comment]
OK. I will revisit it..

**Comment by emminizer** [comment]
If it helps, it looks like this error message only shows up in debug mode, not in release mode (though it is still a problem in release).

**Comment by gwaldron** [comment]
Ok, I updated osgearth master branch with this fix. Let me know if it works for you

**Comment by emminizer** [comment]
Thanks! Yes the update to osgEarth master works great. I'm working on trying to integrate Blend2D into our osgEarth build across all our systems and afterwards will merge this in.

**Pull request merged**